### ISSUE TITLE
Battery Status API implementation shouldn't use impossible values

### ISSUE THREAD

**Issue by thestinger** [bromite/bromite#322]
The current implementation says that the device isn't plugged in, is fully charged and that it's going to immediately run out of power (due to `dischargingTime` being set to 0). If a web application uses the discharging time, it's likely to trigger aggressive power saving modes. Saying that the device isn't plugged in tends to trigger logic to figure out the discharge time.

It would be better to say that the device is plugged in and fully charged, as this is actually reasonable, entirely possible and matches across devices with and without a battery. It's the output given by a phone or laptop that's fully charged and plugged in along with a desktop without a battery. It's the most universal set of values, and won't cause problems.

```c++
bool BatteryManager::charging() {
  return true;
}

double BatteryManager::chargingTime() {
  return 0.0;
}

double BatteryManager::dischargingTime() {
  return std::numeric_limits<double>::infinity();
}

double BatteryManager::level() {
  return 1.0;
}
```

The correct value for `dischargingTime` in the charging or unknown cases is `Infinity`:

https://developer.mozilla.org/en-US/docs/Web/API/BatteryManager/dischargingTime

The value of `charging` is really about whether the device is plugged in and is `true` even if there's no battery, so setting it to `true` is the most universal and then it makes sense that the device is neither charging or discharging when at 100% capacity.

**Comment by csagan5** [comment]
@thestinger you're right, and this additionally provides a tracking vector.

This patch was introduced because https://github.com/Eloston/ungoogled-chromium/blob/62169230d4f48b30bca165b39f3f6f014c6cf57e/patches/extra/inox-patchset/0019-disable-battery-status-service.patch did not seem effective on Android.

I will adjust the values as you proposed.

**Comment by csagan5** [comment]
Fixed in `75.0.3770.109` (not yet released).
